core/tests/protractor_utils/TopicsAndSkillsDashboardPage.js

```diff
@@ -422,25 +372,32 @@ var TopicsAndSkillsDashboardPage = function() {
     await waitFor.elementToBeClickable(
       confirmUnassignSkillButton,
       'Confirm Unassign skill button takes too long to be clickable');
-    await confirmUnassignSkillButton.click();
+    await action.click('Confirm Unassign Skill', confirmUnassignSkillButton);
+    await waitFor.invisibilityOf(
+      confirmUnassignSkillButton,
+      'Unassign skill modal takes too long to close.');
   };
 
   this.navigateToSkillsTab = async function() {
-    await waitFor.elementToBeClickable(
-      skillsTabButton,
-      'Skills tab button taking too long to be clickable');
-    await skillsTabButton.click();
+    await action.click('Skills tab button', skillsTabButton);
   };
 
   this.expectNumberOfTopicsToBe = async function(number) {
-    expect(await topicsListItems.count()).toBe(number);
+    let topicsTableIsPresent = await topicsTable.isPresent();
+    if (topicsTableIsPresent) {
+      expect(await topicsListItems.count()).toBe(number);
+    }
```

Comment:
@kevintab95 why did we do this? It seems like this will let the test pass if the topics table doesn't appear, even if we are expecting topics to be present

Reply:
When there are no skills or topics on the server, we don't show the table. The UI will just show two buttons on the page.

I think a better way would be to use waitFor for the table when the number > 0 and if number is 0, expect that the table is not present.

@U8NWXD